src/tables.jl

```diff
@@ -145,41 +146,55 @@ Load a Tables.jl input `source` into an SQLite table that will be named `tablena
 """
 function load! end
 
-load!(db::DB, table::AbstractString="sqlitejl_"*Random.randstring(5); kwargs...) = x->load!(x, db, table; kwargs...)
+load!(db::DB, name::AbstractString="sqlitejl_"*Random.randstring(5); kwargs...) =
+    x -> load!(x, db, name; kwargs...)
 
 function load!(itr, db::DB, name::AbstractString="sqlitejl_"*Random.randstring(5); kwargs...)
     # check if table exists
-    nm = esc_id(name)
-    db_tableinfo = tableinfo(db, nm)
+    db_tableinfo = tableinfo(db, name)
     rows = Tables.rows(itr)
     sch = Tables.schema(rows)
-    return load!(sch, rows, db, nm, name, db_tableinfo; kwargs...)
+    return load!(sch, rows, db, name, db_tableinfo; kwargs...)
 end
 
-checkdupnames(names) = length(unique(map(x->lowercase(String(x)), names))) == length(names) || error("duplicate case-insensitive column names detected; sqlite doesn't allow duplicate column names and treats them case insensitive")
+# case-insensitive check for duplicate column names
+function checkdupnames(names::Union{AbstractVector, Tuple})
```

Comment:
It still throws in the same situations. However, now it throws `SQLiteException` instead of generic `ErrorException`.
Plus the message indicates the duplicated column, which is quite useful :)